### DISCUSSION TITLE
Run a qutebrowser command from a Greasemonkey script

### DISCUSSION THREAD

**Discussion by michaeljmcd** [qutebrowser/qutebrowser#6351] [Q&A]
Is there a way to run a qutebrowser EX-style command from within a Greasemonkey script? I haven't found anything about it one way or the other.

**Comment by The-Compiler** [comment]
No - Greasemonkey scripts essentially run with the same privileges than websites, so it's likely impossible to allow for this in a secure way. Also see #6264 for some discussion about this.

What's your use-case?

**Comment by michaeljmcd** [comment]
I was hoping to write a quick saver for TiddlyWiki in Qutebrowser. The user
script could easily save the new wiki file, but the content of a part of
the page is needed to make it work.

You can see code for this in the dedicated desktop app:
https://github.com/Jermolene/TiddlyDesktop/blob/master/source/js/utils/saving.js

**Reply by The-Compiler** [reply]
> but the content of a part of the page is needed to make it work.

So you'll need e.g. the selected text of the current page, but as HTML? i.e. I assume `QUTE_SELECTED_TEXT` or `QUTE_HTML` [for userscripts](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc) (unrelated to Greasemonkey scripts) won't work?

**Comment by michaeljmcd** [comment]
I'm not sure. The way the TiddlyWiki desktop saver works is that the new
content of the wiki is loaded into a temporary div that is dynamically
created and the desktop app code reaches in, grabs the contents of the div
and dumps it into the file location.

A userscript would work fine for the raw file access, but it would need
access to the DOM as modified by the page. I would assume (perhaps
incorrectly) that the path indicated by QUTE_HTML refers to the HTML as
originally downloaded. If so, it wouldn't have the updated content to save
back out.

The other consideration is triggering this workflow. The saver above
listens to a DOM event to trigger the save process. So even if QUTE_HTML
had the updated DOM, there would still need to be a way to automatically
invoke the userscript without requiring a manual keybinding (TiddlyWiki
already supports a manual save so an alternate manual save doesn't really
advance the status quo) which is where the thought of Greasemonkey ->
Userscript came in.
